Adding LTI-tools configuration to exclude Live Streams in the LTI-Series-Tool

ciegler opened the pull request:
This PR adds a configuration file for the LTI-tools `etc/ui-config/mh_default_org/ltitools/config.json` and the required changes to exclude Live Streams in the LTI-series-tool when this is set in the configuration file. The configuration file includes the Boolean value `excludeLiveStreams` , which is by default false. When set to true Live Streams are filtered out and not shown in the LTI-series-Tools.

ciegler committed:
Adding a config file for the LTI-Tools
Adding a config file for the LTI-Tools which offers the option to exclude livestreams in the LTI-Series-Tools.

KatrinIhler commented:
Test keeps failing, maybe this indeed breaks something or the test needs to be updated. @ciegler Can you take another look at this when you have time?

lkiesow commented on modules/lti/src/OpencastRest.ts:
```diff
@@ -224,7 +243,8 @@ export async function searchEpisode(
                 tracks: parseTracksFromResult(result)
-            }
+            },
+            isLive: result.mediapackage?.media?.track?.some((track: any) => track.live)
```
I think this line is breaking the tests.
Launching `selenium-tests gui`, you see the error:
```
Error: _result$mediapackage$2.some is not a function
```
This will also break in production depending on the data source, I think.

ciegler replied:
Thank you for pointing that out, I was not sure where the error was coming from. By that I also realized that I could run into cases where the `SearchEpisodeResult.isLive` is undefined, which I do not want, so I also changed that.

ciegler commented:
Sorry for the commit chaos, I just found out that I can actually squash several commits into one. In future I will do this before creating a PR, so that the commit is then laid out clearer.

lkiesow commented on modules/lti/src/OpencastRest.ts:
```diff
@@ -224,7 +243,8 @@ export async function searchEpisode(
                 tracks: parseTracksFromResult(result)
-            }
+            },
+            isLive: result.mediapackage.media.track !== undefined ? result.mediapackage.media.track.isLive : false
```
I haven't actually tested this, but I'm not sure if this will always work. It's quite annoying, but `track` may be a JS object *or* an array. That's why there are all the weird checks above. I think code will not work if there are multiple tracks (i.E. `track` is an array).

KatrinIhler commented:
Apparently Martin has also found a problem while testing this, he will comment the details later this week.

ciegler committed:
Adding parseLiveFromResults to handle that track can be an object or an array

majosch commented:
@ciegler Thanks for the update! Just tested this on a fresh `r/17.x` and works as expected :tada: 

[Bildschirmaufzeichnung vom 2024-12-13 11-21-10.webm](https://github.com/user-attachments/assets/f2241c1b-9102-4381-9a98-87abcc85b12b)

KatrinIhler commented:
@lkiesow Have your concerns been sufficiently addressed? :)

KatrinIhler commented:
@lkiesow Ping :D

majosch commented:
I just noticed, that series with more than 15 events are broken, because the navigation displays only one page.

When setting `excludeLiveStreams` to false, the navigation behaves as expected.

[Bildschirmaufzeichnung vom 2025-02-21 11-33-55.webm](https://github.com/user-attachments/assets/f970670a-4b72-4153-884e-f103d669dbde)

ciegler committed:
Changed wrong  value in filtered SearchEpisodeResults

ciegler commented:
Thanks for caching the bug. I changed the `total` value for the `SearchEpisodeResults` in the original commit and reverted these changes now. This should fix the bug in the navigation.

ciegler commented:
Since filtering out the live episodes in the LTI frontend after the request to the `search index` lead to a wrong total page and total Episodes count, it was requested to implement the possibility to filter out live events already in the search index. Therefore I also had to integrate the configuration file differently. I opened a new PR with the newest changes (https://github.com/opencast/opencast/pull/7032) and will close the PR now.
